hoard/client.py

```diff
@@ -53,5 +55,19 @@ class DSpaceClient:
 
 
 class OAIClient:
-    def __init__(self, url):
-        self.url = url
+    def __init__(self, source_url: str, format: str, set: str = None) -> None:
+        self.source_url = source_url
+        self.format = format
+        self.set = set
+
+    def get(self) -> Generator[int, float, str]:
```

Comment:
Rather than defining a `get()` method, I think Mike was suggesting that the OAIClient class itself be an iterator, so you would define the `__iter__()` and `__next__()` methods, and `__next__()` would return a single record. @gravesm is that correct?

Reply:
Yeah, that's correct. I'd also suggest moving the sickle client creation into the `__init__()` method and giving it a more descriptive name like `client`.